circuits/circom/tallyVotes.circom

```diff
@@ -232,32 +226,28 @@ template ResultCommitmentVerifier(voteOptionTreeDepth) {
         currentResultsRoot.leaves[i] <== currentResults[i];
     }
 
-    component currentResultsCommitment = HashLeftRight();
-    currentResultsCommitment.left <== currentResultsRoot.root;
-    currentResultsCommitment.right <== currentResultsRootSalt;
+    var currentResultsCommitmentHash = PoseidonHash(2)([currentResultsRoot.root, currentResultsRootSalt]);
 
     // Compute the commitment to the current spent voice credits
-    component currentSpentVoiceCreditsCommitment = HashLeftRight();
-    currentSpentVoiceCreditsCommitment.left <== currentSpentVoiceCreditSubtotal;
-    currentSpentVoiceCreditsCommitment.right <== currentSpentVoiceCreditSubtotalSalt;
+    var currentSpentVoiceCreditsCommitmentHash = PoseidonHash(2)([currentSpentVoiceCreditSubtotal, currentSpentVoiceCreditSubtotalSalt]);
 
     // Compute the root of the spent voice credits per vote option
     component currentPerVOSpentVoiceCreditsRoot = QuinCheckRoot(voteOptionTreeDepth);
     for (var i = 0; i < numVoteOptions; i++) {
         currentPerVOSpentVoiceCreditsRoot.leaves[i] <== currentPerVOSpentVoiceCredits[i];
     }
 
-    component currentPerVOSpentVoiceCreditsCommitment = HashLeftRight();
-    currentPerVOSpentVoiceCreditsCommitment.left <== currentPerVOSpentVoiceCreditsRoot.root;
-    currentPerVOSpentVoiceCreditsCommitment.right <== currentPerVOSpentVoiceCreditsRootSalt;
+    var currentPerVOSpentVoiceCreditsCommitmentHash = PoseidonHash(2)([currentPerVOSpentVoiceCreditsRoot.root, currentPerVOSpentVoiceCreditsRootSalt]);
 
     // Commit to the current tally
-    component currentTallyCommitmentHasher = Hasher3();
-    currentTallyCommitmentHasher.in[0] <== currentResultsCommitment.hash;
-    currentTallyCommitmentHasher.in[1] <== currentSpentVoiceCreditsCommitment.hash;
-    currentTallyCommitmentHasher.in[2] <== currentPerVOSpentVoiceCreditsCommitment.hash;
+    var currentTallyCommitmentHash = PoseidonHash(3)([
+        currentResultsCommitmentHash, 
+        currentSpentVoiceCreditsCommitmentHash, 
+        currentPerVOSpentVoiceCreditsCommitmentHash
+    ]);
+
+    // currentTallyCommitmentHash === currentTallyCommitment;
```

Comment:
will we need this in the future?

Reply:
I was gonna suggest too to remove it, to avoid confusion (like it happened ytd to us @0xjei )

Reply:
I left it there to avoid making a mess. If we agree, I would prefer it to be removed in a separate PR so that it can be tracked if it is useful in the future.